api/utils/keys/privatekey_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keys
+
+import (
+	"crypto/tls"
+	"testing"
+
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+)
+
+// TestParsePrivateKey tests that ParsePrivateKey successfully parses private key PEM.
+func TestParsePrivateKey(t *testing.T) {
+	for _, tt := range []struct {
+		desc        string
+		keyPEM      []byte
+		assertError require.ErrorAssertionFunc
+		expectType  interface{}
+	}{
+		{
+			desc:   "invalid PEM",
+			keyPEM: []byte(`non-pem data`),
+			assertError: func(t require.TestingT, err error, i ...interface{}) {
+				require.Error(t, err)
+				require.True(t, trace.IsBadParameter(err))
```

Comment:
nit: Technically, the first assertion is not needed as `nil` is not a bad parameter error. I'm not sure tho, if the error message will be clear enough if you remove it.

Reply:
Yep, I usually do this to get a better error message. Alternatively, we could do `require.True(t, trace.IsBadParameter(err), "expected trace.BadParameter, got %T", err)`